File: frontend/src/app/models/TeacherStudentGradingDTO.ts
export interface UserListingDTO {
    identifier: string;
    firstName: string;
    lastName: string;
    name: string;
  }
  
export interface GradeDTO {
    id: number;
    gradeValue: number;
    gradedBy: string;
    creationDate: string;
}

export interface TeacherStudentGradingDTO {
    courseDetailName: string;
    courseDateId: number;
    courseDateName: string;
    students: UserListingDTO[];
    grades: GradeDTO[];
}

File: backend/src/main/java/com/szakdoga/backend/courses/dtos/TeacherStudentGradingDTO.java
package com.szakdoga.backend.courses.dtos;


import com.szakdoga.backend.auth.dtos.UserListingDTO;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class TeacherStudentGradingDTO {
    private String courseDetailName;
    private Long courseDateId;
    private String courseDateName;
    private List<UserListingDTO> students;
    private List<GradeDTO> grades;
}
